client/src/postItem/postItem.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react';
+import Button from '@material-ui/core/Button';
+import Step1 from './step1.jsx';
+import Step2 from './step2.jsx';
+import Step3 from './step3.jsx';
+import Step4 from './step4.jsx';
+import Step5 from './step5.jsx';
+
+class PostItem extends Component {
+  constructor (props) {
+    super (props);
+    this.state = {
+      step: 0,
+      title: '',
+      category: '',
+      brand: '',
+      model: '',
+      description: '',
+      price: 0,
+      nameYourOwnPrice: false,
+      minimunAcceptedPrice: 0,
+      pickUpLocation: '',
+      availableFrom: '',
+      availableTo: ''
+    };
+    this.changeToPrevious = this.changeToPrevious.bind(this);
+    this.changeToNext = this.changeToNext.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  changeToPrevious () {
+    const { step } = this.state;
+    this.setState ({
+      step: step - 1
+    });
+  }
+
+  changeToNext () {
+    const { step } = this.state;
+    this.setState ({
+      step: step + 1
+    });
+  }
+
+  handleChange (input) {
+    return (e)=> {
+      if (input === 'nameYourOwnPrice') {
+        this.setState ({ [input]: e.target.checked });
```

Comment:
> Is this using destructuring for the key? I've seen a key in square brackets before but I don't know what it means

Reply:
It was the short for initializing an empty object. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names

Reply:
Oh that's very useful, thanks